peps/pep-0724.rst

```diff
@@ -83,6 +83,19 @@ The following code sample demonstrates both of these limitations.
 return ``TypeGuard`` type was not a subtype of the input type. Refer to
 :pep:`647` for examples.
 
+Rationale
+=========
+
+There are a number of issues where a stricter ``TypeGuard`` would have
+been a solution:
+
+* `Python typing issue <https://github.com/python/typing/issues/1351>`__
+* `Pyright issue <https://github.com/microsoft/pyright/issues/3450>`__
+* `Pyright issue <https://github.com/microsoft/pyright/issues/3466>`__
+* `Mypy issue <https://github.com/python/mypy/issues/15305>`__
+* `Mypy issue <https://github.com/python/mypy/issues/15520>`__
+* `Typeshed issue <https://github.com/python/typeshed/issues/8009>`__
```

Comment:
There are now two of each of "pyright issue" and "mypy issue" -- would it be worth disambiguating at all to e.g. pick out key elements of these discussions, or is this just intended as a list of examples where this PEP would be useful?

A

Reply:
I think it's worth being a bit more precise, e.g. `typeshed issue around inspect.isawaitable` or similar